tests/scripts/hunt-filter/exemptions/environments/iceFortress.spec.ts

```diff
@@ -0,0 +1,111 @@
+import {IntakeRejectionEngine} from '@scripts/hunt-filter/engine';
+import {IceFortressStager} from '@scripts/modules/stages/environments/iceFortress';
+import {User} from '@scripts/types/hg';
+import {IntakeMessage} from '@scripts/types/mhct';
+import {LoggerService} from '@scripts/util/logger';
+import {getDefaultIntakeMessage, getDefaultUser} from '@tests/scripts/hunt-filter/common';
+
+describe('Ice Fortress exemptions', () => {
+    let logger: LoggerService;
+    let stager: IceFortressStager;
+    let target: IntakeRejectionEngine;
+
+    beforeEach(() => {
+        logger = {} as LoggerService;
+        stager = new IceFortressStager();
+        target = new IntakeRejectionEngine(logger);
+
+        logger.debug = jest.fn();
+    });
+
+    describe('validateMessage', () => {
+        let preUser: User;
+        let postUser: User;
+        let preMessage: IntakeMessage;
+        let postMessage: IntakeMessage;
+
+        beforeEach(() => {
+            preUser = {...getDefaultUser(), ...getIceFortressUser()};
+            postUser = {...getDefaultUser(), ...getIceFortressUser()};
+            preMessage = {...getDefaultIntakeMessage()};
+            postMessage = {...getDefaultIntakeMessage()};
+        });
+
+        it('should accept when no transitions', () => {
+            preUser.quests.QuestIceFortress = {shield : {is_broken: false},
+                cannons:    {
```

Comment:
Since these are all the same in all the tests, I would just pull this into the default `getIceFortressUser()` then change what is needed. Which is usually nothing in these cases.